ui-v2/app/routes/dc/services/instance.js

```diff
@@ -0,0 +1,29 @@
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+import { get } from '@ember/object';
+
+export default Route.extend({
+  repo: service('repository/service'),
+  proxyRepo: service('repository/proxy'),
+  model: function(params) {
+    const repo = get(this, 'repo');
+    const proxyRepo = get(this, 'proxyRepo');
+    const dc = this.modelFor('dc').dc.Name;
+    return hash({
+      item: repo.findInstanceBySlug(params.id, params.name, dc),
+    }).then(function(model) {
+      return hash({
+        proxy:
+          get(service, 'Kind') !== 'connect-proxy'
+            ? proxyRepo.findInstanceBySlug(params.id, params.name, dc)
+            : null,
+        ...model,
+      });
+    });
+  },
+  setupController: function(controller, model) {
+    this._super(...arguments);
```

Comment:
I don't know if you still want to call super here - pretty sure just does `controller.set('model', model)` so now you'll have `model.item`, `model.proxy`, `item`, and `proxy` in your template context.

Reply:
Hey @meirish 

Sorry, I forgot about this and merged! Agree that it's questionable whether `super` should be called here or not.

I'm going to look into this a little further. I'm not sure if it only sets variables on the controller for you. I vaguely remember reading a blog article saying you should always call `super` in `setupController` otherwise things can go wrong - but I might be misremembering the method it was talking about, pretty sure it was this though as I remember a while ago going through add adding these `super`s in everywhere in places that I'd missed.

Up until now I've found it really difficult to know whether something in ember is deemed a 'hook' (and therefore a super call isn't needed) and when something is 'accepted overriding of a method' and therefore a `super` call is definitely needed. I've mostly just been using super everywhere unless its very clear that its a hook (for example in the `model` hook). I also checked to see what effect calling `super` has when a method on the super class doesn't exist, and as far as I remember its essentially a noop.

Anyway lemme look into it further.

Thanks

Reply:
Ok found the blog post:

https://www.toptal.com/emberjs/the-8-most-common-ember-js-developer-mistakes#common-mistake-no-3-not-calling-the-default-implementation-in-setupcontroller

But, reading that more carefully and looking at the ember source:

https://github.com/emberjs/ember.js/blob/30137796af42c63b28ead127cba0e43e45a773c1/packages/%40ember/-internals/routing/lib/system/route.ts#L1268-L1273

The 'things can go wrong' is that a model variable isn't set in the template context/controller.

In my case I never use `model` to access things so I could just remove this? I can do this in another PR.

Thanks,

Reply:
Added a PR here https://github.com/hashicorp/consul/pull/5383